bossutils/test/test_vault.py

```diff
@@ -61,4 +62,13 @@ def test_exception_if_key_dosent_exist(self, mockClient):
         instance.read.return_value = { "data": {} }
         v = Vault(self.cfg)
         with self.assertRaises(Exception):
-            v.read('secrets', 'super')
\ No newline at end of file
+            v.read('secrets', 'super')
+
+    @patch.object(Vault, 'login')
+    def test_retry_logic(self, mockLogin, mockClient):
+        v = Vault(self.cfg)
+        login = mockLogin.return_value
+        login.login.return_value = True
+        instance = mockClient.return_value
+        instance.read.side_effect = [hvac.exceptions.Forbidden('Token has expired'),v.read('secrets','super')]
```

Comment:
Normally you would use the actual data returned from `read()` instead of a call to `read()`
You are also mixing read methods. `instance.read` is the `hvac` library read while `v.read` is the read of our Vault client.